apps/storefront/src/pages/OrderDetail/components/OrderBilling.tsx

```diff
@@ -1,17 +1,36 @@
-import { useContext } from 'react';
+import { useCallback, useContext, useEffect, useState } from 'react';
+import { InsertDriveFileOutlined } from '@mui/icons-material';
 import { Box, Card, CardContent, Stack, Typography } from '@mui/material';
+import { v1 as uuid } from 'uuid';
 
 import { B3ProductList } from '@/components';
+import B3Dialog from '@/components/B3Dialog';
+import CustomButton from '@/components/button/CustomButton';
 import { useMobile } from '@/hooks';
 import { useB3Lang } from '@/lib/lang';
+import { getDigitalDownloadElements } from '@/shared/service/b2b/graphql/orders';
+import { snackbar } from '@/utils';
+import { getBlobFileName } from '@/utils/getBlobFileName';
+import { handleBlobDownload } from '@/utils/handleBlobDownload';
 
-import { OrderBillings } from '../../../types';
+import { OrderBillings, OrderProductItem, ProductItem } from '../../../types';
 import { OrderDetailsContext } from '../context/OrderDetailsContext';
 
 type OrderBillingProps = {
   isCurrentCompany: boolean;
 };
 
+type DigitalProductNode = {
+  downloadFileUrls: string[];
+  downloadPageUrl: string;
+  name: string;
+  productEntityId: number;
+};
+
+type DigitalProduct = {
+  node: DigitalProductNode;
+};
```

Comment:
If we move these data definitions to the fetch method, we don't need to manually annotate every time we read the array.

```tsx
type DigitalProductNode = {
  downloadFileUrls: string[];
  downloadPageUrl: string;
  name: string;
  productEntityId: number;
};

type DigitalProduct = {
  node: DigitalProductNode;
};

export const getDigitalDownloadElements = (orderId: number | string): Promise<DigitalProduct[]> =>
```